hudi-cli/src/main/java/org/apache/hudi/cli/commands/DiffCommand.java

```diff
@@ -0,0 +1,196 @@
+      @CliOption(key = {"startTs"}, help = "start time for compactions, default: now - 10 days") String startTs,
+      @CliOption(key = {"endTs"}, help = "end time for compactions, default: now - 1 day") String endTs,
+      @CliOption(key = {"limit"}, help = "Limit compactions", unspecifiedDefaultValue = "-1") final Integer limit,
+      @CliOption(key = {"sortBy"}, help = "Sorting Field", unspecifiedDefaultValue = "") final String sortByField,
+      @CliOption(key = {"desc"}, help = "Ordering", unspecifiedDefaultValue = "false") final boolean descending,
+      @CliOption(key = {"headeronly"}, help = "Print Header Only", unspecifiedDefaultValue = "false") final boolean headerOnly,
+      @CliOption(key = {"includeArchivedTimeline"}, help = "Include archived commits as well",
+          unspecifiedDefaultValue = "false") final boolean includeArchivedTimeline) throws IOException {
+    HoodieDefaultTimeline timeline = getTimelineInRange(startTs, endTs, includeArchivedTimeline);
+    return printCommitsWithMetadataForPartition(timeline, limit, sortByField, descending, headerOnly, "", partitionPath);
+  }
+
+  private HoodieDefaultTimeline getTimelineInRange(String startTs, String endTs, boolean includeArchivedTimeline) {
+    if (StringUtils.isNullOrEmpty(startTs)) {
+      startTs = getTimeDaysAgo(10);
+    }
+    if (StringUtils.isNullOrEmpty(endTs)) {
+      endTs = getTimeDaysAgo(1);
+    }
+    HoodieTableMetaClient metaClient = HoodieCLI.getTableMetaClient();
+    HoodieActiveTimeline activeTimeline = metaClient.getActiveTimeline();
+    if (includeArchivedTimeline) {
+      HoodieArchivedTimeline archivedTimeline = metaClient.getArchivedTimeline();
+      archivedTimeline.loadInstantDetailsInMemory(startTs, endTs);
+      return archivedTimeline.findInstantsInRange(startTs, endTs).mergeTimeline(activeTimeline);
+    }
+    return activeTimeline;
+  }
+
+  private String printCommitsWithMetadataForFileId(HoodieDefaultTimeline timeline,
+                                                   final Integer limit,
+                                                   final String sortByField,
+                                                   final boolean descending,
+                                                   final boolean headerOnly,
+                                                   final String tempTableName,
+                                                   final String fileId) throws IOException {
+    final List<Comparable[]> rows = new ArrayList<>();
+
+    final List<HoodieInstant> commits = timeline.getCommitsTimeline().filterCompletedInstants()
+        .getInstants().sorted(HoodieInstant.COMPARATOR.reversed()).collect(Collectors.toList());
+
+    for (final HoodieInstant commit : commits) {
+      if (timeline.getInstantDetails(commit).isPresent()) {
+        final HoodieCommitMetadata commitMetadata = HoodieCommitMetadata.fromBytes(
+            timeline.getInstantDetails(commit).get(),
```

Comment:
this getInstantDetails call for merged timeline is expensive right? avoid double invocation?